### PR TITLE
chore(build): add github workflow for pd test build deploy

### LINKED ISSUE #7064: Create PD test build deploy workflow
# Overview

We need to create a test/build/deploy github workflow (similar to the one for the runapp) so that we can move branch/tag deployments from travis to github actions.

# Implementation details

Run unit tests, e2e tests, then build and deploy to s3.

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview
This PR adds a test + build + deploy github workflow for PD (quite similar to the one for the run app). It also removes PD sandbox deploys from travis and deletes the now unnecessary pd e2e test workflow.

Closes #7064 

# Changelog
- Add a test + build + deploy workflow for PD

# Review requests
Code review

# Risk assessment
Low

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
Approach LGTM!

**Review by mcous — COMMENTED** [review]
Looks good, my only comment is that it might be better to treat the artifact build as a parallel sibling to tests and gate the _deploy_ on tests instead of the artifact

File: .github/workflows/pd-test-build-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+          make -C protocol-designer
+      - name: 'upload github artifact'
+        uses: actions/upload-artifact@v2
+        with:
+          name: 'pd-artifact'
+          path: protocol-designer/dist
+  deploy-pd:
+    name: 'deploy PD artifact to S3'
+    runs-on: 'ubuntu-latest'
+    needs: build-PD
```
Comment (mcous) [inline .github/workflows/pd-test-build-deploy.yaml]: Oops, forgot to actually submit this comment. I think this should be `["js-unit-test", "e2e-test", "build-PD"]`, and the `build-PD` step should have no `needs`

File: .github/workflows/pd-test-build-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+          restore-keys: |
+            js-${{ secrets.GH_CACHE_VERSION }}-${{ runner.os }}-yarn-
+      - name: 'setup-js'
+        run: |
+          npm config set cache ./.npm-cache
+          yarn config set cache-folder ./.yarn-cache
+          make setup-js
+      - name: 'test-e2e'
+        run: make -C protocol-designer test-e2e
+  build-PD:
```
Comment (mcous) [inline .github/workflows/pd-test-build-deploy.yaml]: Might be easier to leave this job name as all lower case. I think GH Actions ignores string casing but why risk it

**Review by mcous — APPROVED** [review]
🎥

**Pull request merged**